worm_war/sources/board.cpp

```diff
@@ -1,10 +1,14 @@
 #include "board.h"
-
+#include <thread>
+#include <mutex>
 
 Board::Board(int width, int height) : nextId_{1}, killAll_{false} {
+    if (height <= 0 || width <= 0 ) {
+        throw std::invalid_argument("Incorrect height or width value.");
+    }
   board_.resize(height);
   for (int i = 0; i < (int)board_.size(); ++i)
-    board_[i].resize(width);
+        board_[i].resize(width);
```

Comment:
We usually try to keep one style, I created whole code using two-space indentation. It's a nit change, so don't fix this in the whole code, just remember for future projects.